litr/src/main/java/com/linkedin/android/litr/render/PassthroughSoftwareRenderer.java

```diff
@@ -42,8 +50,35 @@ public PassthroughSoftwareRenderer(@NonNull Encoder encoder, long frameWaitTimeo
         this.frameWaitTimeoutUs = frameWaitTimeoutUs;
     }
 
+    private int getSampleRate(MediaFormat format) {
+        return format.getInteger(MediaFormat.KEY_SAMPLE_RATE);
+    }
+
     @Override
-    public void init(@Nullable Surface outputSurface, @Nullable MediaFormat sourceMediaFormat, @Nullable MediaFormat targetMediaFormat) {}
+    public void init(@Nullable Surface outputSurface, @Nullable MediaFormat sourceMediaFormat,
+            @Nullable MediaFormat targetMediaFormat) {
+        onMediaFormatChanged(sourceMediaFormat, targetMediaFormat);
+    }
+
+    @Override
+    public void onMediaFormatChanged(@Nullable MediaFormat sourceMediaFormat, @Nullable MediaFormat targetMediaFormat) {
+        this.sourceAudioFormat = sourceMediaFormat;
+        this.targetAudioFormat = targetMediaFormat;
+        initAudioResampler();
+    }
+
+    private void initAudioResampler() {
+        if (sourceAudioFormat == null || targetAudioFormat == null) {
+            return;
+        }
+        int inputSampleRate = getSampleRate(sourceAudioFormat);
```

Comment:
Can't we use `AudioResampler` method here? It seems to be doing the same thing.

Reply:
done